spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -230,8 +230,37 @@ describe("Router", function()
         headers = { ["Host"] = "preserved.com" },
       })
 
-      local body = assert.res_status(404, res)
-      assert.matches("Server at preserved.com", body, nil, true)
+      assert.res_status(200, res)
+      local json = cjson.decode((res:read_body()))
+      assert.equal("preserved.com", json.headers.Host)
+    end)
+
+    pending("preserves downstream host+port if enabled", function()
+      -- pending because httpbin.org seems to not return the exact header
+      -- but a parsed one, with the port stripped.
+      local res = assert(client:send {
+        method = "GET",
+        path = "/get",
+        headers = { ["Host"] = "preserved.com:80" },
+      })
+
+      assert.res_status(200, res)
+      local json = cjson.decode((res:read_body()))
+      assert.equal("preserved.com:80", json.headers.Host)
+    end)
+
+    pending("preserves downstream host+non_default_port if enabled", function()
```

Comment:
as per comment: httpbin does not return the host header, but a parsed version with the port stripped, hence cannot test it...